Wangscape/tilegen/CoordinatePacker.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <assert.h>
+
+namespace tilegen
+{
+
+template<typename I>
+class CoordinatePacker
+{
+public:
+    CoordinatePacker(I size_);
```

Comment:
Probably it should be `is_integral`, because the resulting packed coordinates are not unique when floating-point values are used.